submissions/pozzitive11/frogger-game/app.js

```diff
@@ -0,0 +1,111 @@
+const tileSize = {
+  width: 101,
+  height: 85,
+};
+
+const boardSize = {
+  top: 0,
+  right: tileSize.width * 5,
+  bottom: tileSize.height * 6,
+  left: 0,
+};
+class Enemy {
+  constructor(y, speed) {
+    this.sprite = "images/enemy-bug.png";
+    this.x = 0;
+    this.y = y;
+    this.speed = speed;
+    this.enemyCenter = 35;
+  }
+
+  render() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+  }
+
+  collision() {
+    if (
+      this.x + tileSize.width > player.x &&
+      this.x < player.x &&
+      this.y > player.y &&
+      this.y < player.y + tileSize.height
+    ) {
+      player.moveToStartPosition();
+    }
+  }
+
+  update(dt) {
+    this.x += this.speed * dt;
+    if (this.x > boardSize.right) {
+      this.x = -tileSize.width;
+    }
+    this.collision();
+  }
+}
+class Player {
+  constructor() {
+    this.moveToStartPosition();
+    this.sprite = "images/char-boy.png";
+  }
+
+  moveToStartPosition() {
+    this.x = tileSize.width * 2;
+    this.y = tileSize.height * 5 - tileSize.width / 2;
+  }
+
+  resetToStartPosition() {
+    if (this.y < boardSize.top) {
+      this.moveToStartPosition();
+    }
+  }
+
+  render() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+  }
```

Comment:
ENemy class has exactly the same method. Let's make this [DRY](https://www.digitalocean.com/community/tutorials/what-is-dry-development). Consider having a base class for both Player and Enemy, that would handle all shared properties and behaviour.